### ISSUE TITLE
MinLen type refinement in for loop

### ISSUE THREAD

**Issue by as56** [kelloggm/checker-framework#138]
Consider the following code:
```
import org.checkerframework.checker.index.qual.*;

public class newbug {
	public void m1(int [] a) {
		for(int i = 0; i < a.length; i++) {
			int @MinLen(1)[] b = a; // checker warning
		}
	}
	//Just as a reference, this case does not result in a warning
	public void m2(int [] a) {
		if(a.length > 0) {
			int @MinLen(1)[] b = a;	// no warning
		}	
	}
}
```

Please note that the line inside of the for loop in m1 is only executed if the array is @MinLen(1). The index checker however produces a warning in this case. As a reference I have included m2 where the checker works correctly and issues no warning.

Index Checker Warning:
newbug.java:6: error: [assignment.type.incompatible] incompatible types in assignment.
			int @MinLen(1)[] b = a; // checker warning
			                     ^
  found   : @MinLen(0) int @MinLen(0) []
  required: @MinLen(0) int @MinLen(1) []
1 error

**Comment by mernst** [comment]
I agree this is a false positive.  Does it come up in realistic code?  (We don't want to expand the Index Checker to verify arbitrary code, because each extension makes it slower and makes its implementation harder to understand.)

**Comment by mernst** [comment]
The same issue came up recently in Daikon.
